Regression: Lazy compilation broken after v5.26.0

# Bug report

**What is the current behavior?**

When enabling lazy compilation webpack crashes when it first tries to compile a lazy import.

```bash
/███/██████/node_modules/webpack/lib/NormalModule.js:1049
[0]             return this.generator.getTypes(this);
[0]                                   ^
[0]
[0] TypeError: Cannot read property 'getTypes' of undefined
[0]     at NormalModule.getSourceTypes (/███/██████/node_modules/webpack/lib/NormalModule.js:1049:25)
[0]     at addModuleToChunksInfoMap (/███/██████/node_modules/webpack/lib/optimize/SplitChunksPlugin.js:1119:34)
[0]     at /███/██████/node_modules/webpack/lib/optimize/SplitChunksPlugin.js:1193:9
[0]     at Hook.eval [as call] (eval at create (/███/██████/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:13:16)
[0]     at Hook.CALL_DELEGATE [as _call] (/███/██████/node_modules/tapable/lib/Hook.js:14:14)
[0]     at Compilation.seal (/███/██████/node_modules/webpack/lib/Compilation.js:2292:36)
[0]     at /███/██████/node_modules/webpack/lib/Compiler.js:1078:20
[0]     at /███/██████/node_modules/webpack/lib/Compilation.js:2116:4
[0]     at Hook.eval [as callAsync] (eval at create (/███/██████/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:24:1)
[0]     at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/███/██████/node_modules/tapable/lib/Hook.js:18:14)
[0] npm ERR! code 1
[0] npm ERR! path /███/██████/
[0] npm ERR! command failed
[0] npm ERR! command sh -c webpack serve
```

The client side error returned is:

```
Error: Problem communicating active modules to the server: undefined undefined:undefined:undefined undefined
    at ██/████.js:3570:6
    at Set.forEach (<anonymous>)
    at EventSource.activeEventSource.onerror (██/████..js:3568:18)
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Enable lazy compilation in latest webpack version and do a dynamic module `import()`

**What is the expected behavior?**

Webpack should not crash

**Other relevant information:**
webpack version:  v5.27.1
Node.js version: v15.x
Operating System: Linux

--- a/lib/hmr/LazyCompilationPlugin.js
+++ b/lib/hmr/LazyCompilationPlugin.js
@@ -56,9 +56,9 @@ const checkTest = (test, module) => {
 const TYPES = new Set(["javascript"]);
 
 class LazyCompilationDependency extends Dependency {
-	constructor(originalModule) {
+	constructor(proxyModule) {
 		super();
-		this.originalModule = originalModule;
+		this.proxyModule = proxyModule;
 	}
 
 	get category() {
@@ -73,7 +73,7 @@ class LazyCompilationDependency extends Dependency {
 	 * @returns {string | null} an identifier to merge equal requests
 	 */
 	getResourceIdentifier() {
-		return this.originalModule.identifier();
+		return this.proxyModule.originalModule.identifier();
 	}
 }
 
@@ -116,6 +116,10 @@ class LazyCompilationProxyModule extends Module {
 	updateCacheModule(module) {
 		super.updateCacheModule(module);
 		const m = /** @type {LazyCompilationProxyModule} */ (module);
+		this.originalModule = m.originalModule;
+		this.request = m.request;
+		this.client = m.client;
+		this.data = m.data;
 		this.active = m.active;
 	}
 
@@ -154,7 +158,7 @@ class LazyCompilationProxyModule extends Module {
 		const dep = new CommonJsRequireDependency(this.client);
 		this.addDependency(dep);
 		if (this.active) {
-			const dep = new LazyCompilationDependency(this.originalModule);
+			const dep = new LazyCompilationDependency(this);
 			const block = new AsyncDependenciesBlock({});
 			block.addDependency(dep);
 			this.addBlock(block);
@@ -282,7 +286,7 @@ class LazyCompilationDependencyFactory extends ModuleFactory {
 		const dependency = /** @type {LazyCompilationDependency} */ (data
 			.dependencies[0]);
 		callback(null, {
-			module: dependency.originalModule
+			module: dependency.proxyModule.originalModule
 		});
 	}
 }

--- a/lib/hmr/LazyCompilationPlugin.js
+++ b/lib/hmr/LazyCompilationPlugin.js
@@ -32,6 +32,13 @@ const { registerNotSerializable } = require("../util/serialization");
 /** @typedef {import("../util/Hash")} Hash */
 /** @typedef {import("../util/fs").InputFileSystem} InputFileSystem */
 
+const IGNORED_DEPENDENCY_TYPES = new Set([
+	"import.meta.webpackHot.accept",
+	"import.meta.webpackHot.decline",
+	"module.hot.accept",
+	"module.hot.decline"
+]);
+
 /**
  * @param {undefined|string|RegExp|Function} test test option
  * @param {Module} module the module
@@ -340,6 +347,7 @@ class LazyCompilationPlugin {
 						if (
 							resolveData.dependencies.every(
 								dep =>
+									IGNORED_DEPENDENCY_TYPES.has(dep.type) ||
 									(this.imports && dep.type === "import()") ||
 									(this.entries && dep.type === "entry")
 							) &&

--- a/test/hotCases/lazy-compilation/simple/index.js
+++ b/test/hotCases/lazy-compilation/simple/index.js
@@ -1,16 +1,31 @@
 it("should compile to lazy imported module", done => {
 	let resolved;
 	const promise = import("./module").then(r => (resolved = r));
+	let generation = 0;
+	import.meta.webpackHot.accept("./module", () => {
+		generation++;
+	});
 	expect(resolved).toBe(undefined);
 	setTimeout(() => {
 		expect(resolved).toBe(undefined);
+		expect(generation).toBe(0);
 		NEXT(
 			require("../../update")(done, true, () => {
 				promise.then(result => {
 					expect(result).toHaveProperty("default", 42);
-					setTimeout(() => {
-						done();
-					}, 1000);
+					expect(generation).toBe(0);
+					NEXT(
+						require("../../update")(done, true, () => {
+							expect(result).toHaveProperty("default", 42);
+							expect(generation).toBe(1);
+							import("./module").then(result => {
+								expect(result).toHaveProperty("default", 43);
+								setTimeout(() => {
+									done();
+								}, 1000);
+							}, done);
+						})
+					);
 				}, done);
 			})
 		);

--- a/test/hotCases/lazy-compilation/simple/module.js
+++ b/test/hotCases/lazy-compilation/simple/module.js
@@ -1,1 +1,5 @@
 export default 42;
+---
+export default 42;
+---
+export default 43;

